src/viewer.js

```diff
@@ -124,14 +140,57 @@ class Viewer {
   }
   
   /**
-   * @param {?string} urlPath
+   * @param {boolean} isBack true if back button was hit. false otherwise.
+   * @param {boolean} isLastBack true if back button was hit and viewer should close.
+   * @param {number} opt_stackIndex
    * @private
     */
-  handleChangeHistoryState_(urlPath) {
-    if (urlPath) {
-      if (this.showViewer_) this.showViewer_();
-    } else {
-      if (this.hideViewer_) this.hideViewer_();
+  handleChangeHistoryState_(isBack, isLastBack, opt_stackIndex) {
+    if (isBack) {
+      this.viewerMessaging_.sendRequest(
+        'historyPopped', 
+        {'newStackIndex': opt_stackIndex - 1}, 
```

Comment:
opt_stackIndex is the new stackIndex, right? So you shouldn't need to subtract one.

Can you also add jsdoc to clarify is opt_stackIndex is expected to be old or new stack index